sample/sample.go

```diff
@@ -0,0 +1,196 @@
+package main
```

Comment:
this file is temporary for showing the usage, will be removed in the final version.